diff -ur libtta-c-2.0/filter.h libtta-c-src/filter.h
--- libtta-c-2.0/filter.h	2010-12-23 23:06:14.000000000 -0800
+++ libtta-c-src/filter.h	2011-01-10 15:29:02.000000000 -0800
@@ -134,7 +134,7 @@
 ///////////////////////// hybrid_filter_compat_dec //////////////////////////
 /////////////////////////////////////////////////////////////////////////////
 
-__inline void hybrid_filter_dec(TTA_fltst *fs, TTAint32 *in) {
+void hybrid_filter_dec(TTA_fltst *fs, TTAint32 *in) {
 	register TTAint32 *pA = fs->dl;
 	register TTAint32 *pB = fs->qm;
 	register TTAint32 *pM = fs->dx;
@@ -170,7 +170,7 @@
 ///////////////////////// hybrid_filter_compat_enc //////////////////////////
 /////////////////////////////////////////////////////////////////////////////
 
-__inline void hybrid_filter_enc(TTA_fltst *fs, TTAint32 *in) {
+void hybrid_filter_enc(TTA_fltst *fs, TTAint32 *in) {
 	register TTAint32 *pA = fs->dl;
 	register TTAint32 *pB = fs->qm;
 	register TTAint32 *pM = fs->dx;
diff -ur libtta-c-2.0/libtta.c libtta-c-src/libtta.c
--- libtta-c-2.0/libtta.c	2010-12-30 04:42:54.000000000 -0800
+++ libtta-c-src/libtta.c	2011-01-10 15:29:08.000000000 -0800
@@ -10,7 +10,7 @@
  */
 
 #include "libtta.h"
-#include "config.h"
+#include "../config.h"
 #include "filter.h"
 
 /////////////////////////////// portability /////////////////////////////////
diff -ur libtta-c-2.0/libtta.h libtta-c-src/libtta.h
--- libtta-c-2.0/libtta.h	2010-12-24 00:28:09.000000000 -0800
+++ libtta-c-src/libtta.h	2011-01-10 15:29:13.000000000 -0800
@@ -18,7 +18,6 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h> 
-#include <malloc.h> 
 
 #ifdef CARIBBEAN
 #define ALLOW_OS_CODE 1
