--- libtta-cpp-2.3/libtta.cpp	2015-02-19 08:05:00.000000000 +0100
+++ libtta++-src/libtta.cpp	2020-02-26 19:16:19.000000000 +0100
@@ -25,11 +25,19 @@
 #define tta_malloc RMMalloc
 #define tta_free RMFree
 #else // GNUC
+#ifdef __APPLE__
+#define tta_memclear(__dest,__length) memset(__dest,0,__length)
+#define tta_memcpy(__dest,__source,__length) memcpy(__dest,__source,__length)
+#define tta_malloc(__length) malloc(__length)
+// Apple platform malloc() is 16-byte aligned by default.
+#define tta_free free
+#else
 #define tta_memclear(__dest,__length) memset(__dest,0,__length)
 #define tta_memcpy(__dest,__source,__length) memcpy(__dest,__source,__length)
 #define tta_malloc(__length) aligned_alloc(16,__length)
 #define tta_free free
 #endif
+#endif
 #else // MSVC
 #define tta_memclear(__dest,__length) ZeroMemory(__dest,__length)
 #define tta_memcpy(__dest,__source,__length) CopyMemory(__dest,__source,__length)
